pkg/healthcheck/healthcheck.go

```diff
@@ -222,22 +223,29 @@ func (hc *HealthChecker) allCategories() []category {
 					description: "can initialize the client",
 					hintURL:     "https://linkerd.io/2/faq/#k8s-api",
 					fatal:       true,
-					check: func() (err error) {
+					check: func(_ context.Context) (err error) {
 						hc.kubeAPI, err = k8s.NewAPI(hc.KubeConfig, hc.KubeContext)
-						return
+						if err != nil {
+							return err
+						}
+						// k8s' client-go doesn't support injecting context
+						// https://github.com/kubernetes/kubernetes/issues/46503
+						// but we can set the timeout manually
+						hc.kubeAPI.Timeout = requestTimeout
+						return nil
 					},
 				},
 				{
 					description: "can query the Kubernetes API",
 					hintURL:     "https://linkerd.io/2/faq/#k8s-api",
 					fatal:       true,
-					check: func() (err error) {
+					check: func(ctx context.Context) (err error) {
 						hc.httpClient, err = hc.kubeAPI.NewClient()
 						if err != nil {
-							return
+							return err
 						}
-						hc.kubeVersion, err = hc.kubeAPI.GetVersionInfo(hc.httpClient)
-						return
+						hc.kubeVersion, err = hc.kubeAPI.GetVersionInfo(ctx, hc.httpClient)
+						return err
```

Comment:
recommend either using a named return (`func() (err error)`), or explicitly returning vars (`return err`), but not both. i have a _slight_ preference for explicitly returning vars, but care more that a function only implements one or the other.

a bit more info:
https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

Reply:
Agreed. I missed that what we had originally was named returns. I'll revert it to how it was to avoid increasing the diff with an unnecessary change.